src/js/background.js

```diff
@@ -0,0 +1,194 @@
+import {getAccessToken, getToken, getTokenUrl} from "./solid.js";
+import {createDpopHeader} from '@inrupt/solid-client-authn-core';
+
+var id;
+var secret;
+var tokenUrl;
+
+var isChrome;
+
+
+/**
+ * Main function that is called upon extension (re)start
+ */
+function main() {
+
+    isChrome = (navigator.userAgent.toLowerCase().includes("chrome"));
+
+    getCredentialsFromBrowserStorage();
+
+    /**
+     * Runtime message listener, capable of handling and properly awaiting asynchronous functions
+     */
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        handleMessage(request).then(sendResponse);
+        return true;
+    });
+
+    /**
+     * Blocking web request listener that blocks a web request until the alteration of its request headers is completed
+     */
+    chrome.webRequest.onBeforeSendHeaders.addListener(
+        rewriteRequestHeaders,
+        {
+            urls: ["<all_urls>"]
+        },
+        ["blocking", "requestHeaders"]
+    )
+}
+
+/**
+ * Rewrite request headers with correct credentials so that successful authentication may be achieved
+ * @param {Object} details - Request details passed on from blocking web request listener
+ * @returns {Object} - Object containing altered request headers, to be handled by the web request listener callback
+ */
+async function rewriteRequestHeaders(details) {
+
+    // TODO: find a more elegant way to catch the access token creation request called from getAccessToken()
+    if (details.method === "POST") {
```

Comment:
1) Can you make an issue for this TODO?
2) Can you add to the README that only GET requests are supported. I assume that all POST requests will fail, right?

Reply:
Posts requests from the user will not fail or get blocked, they will just not have any authentication, if that is what you meant by 'fail'

I made an issue for this: #14 
Documentation for this was added to the readme in 5d604c5f83ec276107c7331b4c57c03e12f1a4a9